security.rst

```diff
@@ -1254,6 +1254,16 @@ In the above configuration, users with ``ROLE_ADMIN`` role will also have the
 ``ROLE_USER`` role. The ``ROLE_SUPER_ADMIN`` role has ``ROLE_ADMIN``, ``ROLE_ALLOWED_TO_SWITCH``
 and ``ROLE_USER`` (inherited from ``ROLE_ADMIN``).
 
+.. note::
+
+    The role hierarchy defined in the ``role_hierarchy`` option is static and
+    you cannot override it dynamically (for example to store the role hierarchy
+    in a database). If you need that, create a custom
+    :doc:`security voter </security/voters>` that looks for the user roles in
+    the database. Another solution is to create a service that implements the
```

Comment:
maybe `One solution is ...` instead of `Another ...`? because technically creating your own `RoleHierarchyVoter` is exactly what the previous sentence suggests? :blush:

Reply:
Actually I think its fine as it is :blush: